go/coordinator/internal/memberlist_manager/memberlist_manager.go

```diff
@@ -0,0 +1,96 @@
+package memberlist_manager
+
+import (
+	"fmt"
+
+	"k8s.io/client-go/util/workqueue"
+)
+
+// TODO:
+/*
+- Switch to camel case for all variables and methods
+*/
+
+// A memberlist manager is responsible for managing the memberlist for a
+// coordinator. A memberlist is a CR in the coordinator's namespace that
+// contains a list of all the members in the coordinator's cluster.
+// The memberlist uses k8s watch to monitor changes to pods and then updates a CR
+
+type IMemberlistManager interface {
```

Comment:
The reason I did not is because this interface may grow to cover other things. Go makes this a bit weird because we can't really have a notion of a "subclass"